sphinx_needs/api/need.py

```diff
@@ -641,6 +649,34 @@ def _prepare_template(app: Sphinx, needs_info: NeedsInfoType, template_key: str)
     return new_content
 
 
+# Here we want to have the same signature as nested_parse_with_titles from sphinx,
+# but it looks like something is strange between sphinx and docutils,
+# so we return Any instead of str.
+def render_rst(
+    state: Any, content: StringList, node: nodes.Node, content_offset: int = 0
+) -> Any:
+    if content_offset == 0:
+        return nested_parse_with_titles(state, content, node)
+    else:
+        from inspect import signature
+
+        sig = signature(nested_parse_with_titles)
+        # check if we have a doctuils version which already supports content_offset
+        if "content_offset" in sig.parameters:
+            return nested_parse_with_titles(state, content, node, content_offset)
+        else:
+            # let's reimplement a new nested_parse_with_titles
```

Comment:
I have no Idea what you feedback want's to say. As we do not have a proper implementation with old sphinx version, we have to have our own.

Reply:
Additional, I want to couple it in a function, as with my next PR, I want to reuse it in another file.

Reply:
On second thoughts, we should not be using `nested_parse_with_titles` in the first place, only `nested_parse`; you can only parse titles, i.e. section headings, unless you are at the root of your document, which `Need` nodes are not

Put simply, if you have section headings in the content of your need directive, it would break the docutils document structure, and you will produce incorrect HTML

Reply:
Could we create a test to show `nested_parse_with_titles` is wrong?
I would suggest to evaluate this in another PR.

Reply:
Perhaps @danwos can comment on the design decision behind using `nested_parse_with_titles`?

Reply:
I believe the discussion is an evidence to have our own function to capsulate the rendering of rst content. It is a one time location to change it.

Reply:
> Could we create a test to show nested_parse_with_titles is wrong?

Ok, just for you, I've created a whole "blog" post on why it is intrisically wrong 😏: https://gist.github.com/chrisjsewell/0c5827add50074fef0937e2543e955b4
(I jest, but in seriousness I have had to explain so many times now, within different contexts)

Reply:
I suggest to change this with another PR.

Reply:
Topic has now it's own issue: #1155